lib/src/kvs/ds.rs

```diff
@@ -1232,25 +1232,22 @@ impl Datastore {
 		self.notification_channel.as_ref().map(|v| v.1.clone())
 	}
 
-	#[allow(dead_code)]
-	pub(crate) fn live_sender(&self) -> Option<Arc<RwLock<Sender<Notification>>>> {
-		self.notification_channel.as_ref().map(|v| Arc::new(RwLock::new(v.0.clone())))
+	/// Performs a database import from SQL
+	#[instrument(level = "debug", skip(self, sess, sql))]
+	pub async fn import(&self, sql: &str, sess: &Session) -> Result<Vec<Response>, Error> {
+		// Execute the SQL import
+		self.execute(sql, sess, None).await
 	}
 
 	/// Performs a full database export as SQL
 	#[instrument(level = "debug", skip(self, sess, chn))]
 	pub async fn export(
 		&self,
 		sess: &Session,
-		ns: String,
-		db: String,
 		chn: Sender<Vec<u8>>,
 	) -> Result<impl Future<Output = Result<(), Error>>, Error> {
-		// Skip auth for Anonymous users if auth is disabled
-		let skip_auth = !self.is_auth_enabled() && sess.au.is_anon();
-		if !skip_auth {
-			sess.au.is_allowed(Action::View, &ResourceKind::Any.on_db(&ns, &db))?;
-		}
+		// Retrieve the provided NS and DB
+		let (ns, db) = crate::iam::check::check_ns_db(sess)?;
```

Comment:
if I'm not mistaken, this will become a problem with the auth levels (cc @gguillemas). You may want to authenticate with a root-level user, but export a specific database.

With auth-levels, here the Namespace and the Database will be empty, or potentially pointing to a NS and DB which are different from the DB you want to export

Reply:
my bad, the new auth headers don't touch the session